src/inference/include/openvino/runtime/config.hpp

```diff
@@ -0,0 +1,616 @@
+ * can be passed to ExecutableNetwork::GetMetric.
+ *
+ */
+static constexpr Key<std::vector<std::string>, ConfigMutability::RO> supported_metrics{"SUPPORTED_METRICS"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of supported config keys.
+ *
+ * This can be used as an executable network metric as well.
+ *
+ * Each of the returned device configuration keys can be passed to Core::SetConfig, Core::GetConfig, and
+ * Core::LoadNetwork, configuration keys for executable networks can be passed to ExecutableNetwork::SetConfig and
+ * ExecutableNetwork::GetConfig.
+ *
+ */
+static constexpr Key<std::vector<std::string>, ConfigMutability::RO> supported_config_keys{"SUPPORTED_CONFIG_KEYS"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of optimization options per device.
+ */
+static constexpr Key<std::vector<std::string>, ConfigMutability::RO> optimization_capabilities{
+    "OPTIMIZATION_CAPABILITIES"};
+
+/**
+ * @brief Metric which defines support of import/export functionality by plugin
+ */
+static constexpr Key<bool, ConfigMutability::RO> import_export_support{"IMPORT_EXPORT_SUPPORT"};
+
+/**
+ * @brief Metric to get a name of network
+ */
+static constexpr Key<std::string, ConfigMutability::RO> network_name{"NETWORK_NAME"};
+
+/**
+ * @brief Metric to get an unsigned integer value of optimal number of executable network infer requests.
+ */
+static constexpr Key<uint32_t, ConfigMutability::RO> optimal_number_of_infer_requests{
+    "OPTIMAL_NUMBER_OF_INFER_REQUESTS"};
+
+namespace hint {
+
+/**
+ * @brief Enum to define possible performance measure hints
+ */
+enum class ModelPriority {
+    LOW = 0,
+    MEDIUM = 1,
+    HIGH = 3,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const ModelPriority& model_priority) {
+    switch (model_priority) {
+    case ModelPriority::LOW:
+        return os << "LOW";
+    case ModelPriority::MEDIUM:
+        return os << "MEDIUM";
+    case ModelPriority::HIGH:
+        return os << "HIGH";
+    default:
+        throw ov::Exception{"Unsupported performance measure hint"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, ModelPriority& model_priority) {
+    std::string str;
+    is >> str;
+    if (str == "LOW") {
+        model_priority = ModelPriority::LOW;
+    } else if (str == "MEDIUM") {
+        model_priority = ModelPriority::MEDIUM;
+    } else if (str == "HIGH") {
+        model_priority = ModelPriority::HIGH;
+    } else {
+        throw ov::Exception{"Unsupported model priority: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief High-level OpenVINO model priority hint
+ * Defines what model should be provided with more performant bounded resource first
+ */
+static constexpr Key<ModelPriority> model_priority{"MODEL_PRIORITY"};
+
+/**
+ * @brief Enum to define possible performance measure hints
+ */
+enum class PerformanceMeasure {
```

Comment:
For me the name looks a bit confusion. I assume the `Measure` word is connected to performance measurements and analysis, while this particular enum is used as a hint to plugin regarding the use cases of the model.

Why not just call it `PerformaceHint`? `PerformanceMode`?

Reply:
Will be replaced with `PerformanceMode`